backend/wmg/pipeline/integrated_corpus/extract.py

```diff
@@ -37,33 +24,45 @@ def get_X_raw(anndata_object: anndata.AnnData) -> Union[np.ndarray, sparse.spmat
     return raw_expression_matrix if raw_expression_matrix is not None else anndata_object.X
 
 
-def get_dataset_s3_uris() -> Dict[str, str]:
+def get_dataset_asset_urls(datasets=None) -> Dict[str, str]:
     """
-    Retrieve list of s3 uris for datasets included in the wmg cube
+    Retrieve list of asset urls for datasets included in the wmg cube
+
+    :param datasets: list of datasets to check, if None, will retrieve from API
+        This parameter is used for tests.
     """
-    s3_uris = dict()
-    for dataset in get_business_logic().get_all_mapped_datasets():
-        if (
-            (dataset.metadata is not None)
-            and (dataset.metadata.assay is not None)
-            and (dataset.metadata.is_primary_data == "PRIMARY")
-            and (dataset.metadata.organism is not None)
-            and any(assay.ontology_term_id in INCLUDED_ASSAYS for assay in dataset.metadata.assay)
-            and len(dataset.metadata.organism) < 2
-        ):
-            s3_uri = next(
-                a.uri
-                for a in dataset.artifacts
-                if a.type == DatasetArtifactType.H5AD and a.get_file_name() == "local.h5ad"
-            )
-            s3_uris[dataset.dataset_id.id] = s3_uri
-    return s3_uris
+    # hardcode to dev backend if deployment is rdev or test
+    API_URL = (
+        "https://api.cellxgene.dev.single-cell.czi.technology"
+        if os.environ.get("DEPLOYMENT_STAGE") in ["test", "rdev"]
+        else os.getenv("API_URL")
+    )
+
+    asset_urls = dict()
+    if API_URL:
+        dataset_metadata_url = f"{API_URL}/curation/v1/datasets"
+        if datasets is None:
+            datasets = requests.get(dataset_metadata_url).json()
+
+        for dataset in datasets:
+            if (
+                dataset["organism"] is not None
+                and dataset["assay"] is not None
+                and len(dataset["is_primary_data"]) == 1
+                and dataset["is_primary_data"][0]
+                and any(assay["ontology_term_id"] in INCLUDED_ASSAYS for assay in dataset["assay"])
+                and len(dataset["organism"]) < 2
+            ):
+                dataset_id = dataset["dataset_id"]
+                asset_url = next(a["url"] for a in dataset["assets"] if a["filetype"] == "H5AD")
+                asset_urls[dataset_id] = asset_url
+    return asset_urls
 
 
 def copy_datasets_to_instance(s3_uris: Dict, dataset_directory: str):
     """Copy given list of s3 uris to the provided path"""
     for dataset in s3_uris:
-        copy_command = ["aws", "s3", "cp", s3_uris[dataset], f"./{dataset_directory}/{dataset}/local.h5ad"]
+        copy_command = ["wget", s3_uris[dataset], "-O", f"./{dataset_directory}/{dataset}/local.h5ad"]
```

Comment:
Question for reviewers - is wget okay to use? Any disadvantages here? Curl preferred? Why/why not?

Reply:
https://daniel.haxx.se/docs/curl-vs-wget.html